.github/workflows/lint.yml

```diff
@@ -0,0 +1,30 @@
+name: Lint
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Check code style
```

Comment:
Also add a stage called type check.
Run command `npx nuxi typecheck`
This will ensure codebase has consistent typing. (Typechecks are disabled by default in nuxt codebase to increase dev speed)

Reply:
> Also add a stage called type check. Run command `npx nuxi typecheck`
 
@amandesai01 My thought is let's first merge it. I have some doubt and I have to explore a bit about `npx nuxi typecheck` because I checked docs https://nuxt.com/docs/guide/concepts/typescript#type-checking and we have to require some other dep to have it.

Also, We will be doing it in another PR.

Reply:
while doing this I'm getting some type errors 

```powershell
PS C:\Users\shivam sharma\Desktop\My Project\open-source\vidur> npx nuxi typecheck
ℹ Using default Tailwind CSS file                                                                                          nuxt:tailwindcss 6:59:23 pm
ℹ Using zod with vee-validate                                                                                                               6:59:24 pm
ℹ Running with compatibility version 4                                                                                                      6:59:26 pm
app/components/HandleFlexStrip.vue:24:27 - error TS2551: Property 'handle' does not exist on type 'CreateComponentPublicInstanceWithMixins<Readonly<{ handles?: { key: string; value: string; userId: string | null; }[] | undefined; }> & Readonly<{}>, { handles: typeof handles; getHandleIcon: typeof getHandleIcon; }, ... 22 more ..., {}>'. Did you mean 'handles'?

24       v-if="getHandleIcon(handle)"
                             ~~~~~~

app/components/HandleFlexStrip.vue:29:10 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

29         :name="getHandleIcon(handle)"
            ~~~~


app/composables/auth.ts:1:30 - error TS2307: Cannot find module '~~/shared/types/profile-types' or its corresponding type declarations.

1 import type { Session } from '~~/shared/types/profile-types';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

app/plugins/auth.ts:1:30 - error TS2307: Cannot find module '~~/shared/types/profile-types' or its corresponding type declarations.

1 import type { Session } from '~~/shared/types/profile-types';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

server/utils/user.ts:1:35 - error TS2307: Cannot find module '~~/shared/types/profile-types' or its corresponding type declarations.

1 import type { BasicProfile } from '~~/shared/types/profile-types';
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
did you pull latest version?

Reply:
> did you pull latest version?

Yes. Now everything is perfect locally. let's merge it.